Dependency injection/Hilt.md

```diff
@@ -0,0 +1,178 @@
+
+That includes:
+
+  * Generating components with different scopes to use with your modules - `SingletonComponent`, `ActivityComponent`, `FragmentComponent`, `ViewModelComponent` etc.
+  * Built-in `@ActivityContext` and `ApplicationContext` annotations to make context injection easier
+  * Populate generated components with annotated Activities, Fragments, Modules, ViewModels etc.
+
+### Setup
+
+First, add the `hilt-android-gradle-plugin` plugin to your project's root `build.gradle` file:
+
+```groovy
+buildscript {
+    ...
+    ext.hilt_version = '2.33-beta'
+    dependencies {
+        ...
+        classpath "com.google.dagger:hilt-android-gradle-plugin:$hilt_version"
+    }
+}
+```
+
+Then, apply the Gradle plugin and add these dependencies in your `app/build.gradle` file:
+
+```groovy
+...
+apply plugin: 'kotlin-kapt'
+apply plugin: 'dagger.hilt.android.plugin'
+
+android {
+    ...
+}
+
+dependencies {
+    implementation "com.google.dagger:hilt-android:$hilt_version"
+    kapt "com.google.dagger:hilt-compiler:$hilt_version"
+}
+```
+
+Hilt uses Java 8 features. To enable Java 8 in your project, add the following to the `app/build.gradle` file:
+
+```groovy
+android {
+  ...
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
+  }
+}
+```
+
+### Annotations
+
+Let's start by listing the most important annotations. Later we will expand by explaining how each of them works.
+
+  * `@HiltAndroidApp` - Place above your application class to get the magic going. It generates all previously mentioned components and even some modules that you might need. This one is required for everything to work.
+  * `@AndroidEntryPoint` - This one is placed above Activities and Fragments. After doing so, the generated builder modules will be populated with your screens. This is required for each screen that you want to inject to. If you are using ViewModels in your application, this annotation will make sure that the ViewModel can be injected.
+  * `@Inject` - Placed above or beside the dependency that you want to inject.
+  * `@Module` - Marks a class as a module that provides some dependencies. Each module must also be annotated with `@InstallIn` annotation to specify it's scope.
+  * `@InstallIn` - Takes a single parameter that is a component to install the module in. Components have different scopes so make sure that your module is included only where needed
+  * `@Provides` - Place above the function in module that returns an instance of the dependency that you want to inject somewhere.
+  * `@Binds` - Also used in modules, but the function can be abstract or an interface function. Intead of creating and returning the instance, the function must return dependency type (usually an interface) and take the concrete implementation as a parameter.
```

Comment:
```suggestion
  * `@Binds` - Also used in modules, but the function can be abstract or an interface function. Instead of creating and returning the instance, the function must return a dependency type (usually an interface) and take the concrete implementation as a parameter.
```